### ISSUE TITLE
[BUG] Fail to change backup target from S3 to NFS

### ISSUE THREAD

**Issue by janeczku** [harvester/harvester#1791]
harvester 1.0.0

**To Reproduce**
1. Configure S3 backup target
2. Edit backup target and configure NFS target

**Expected behavior**
NFS backup target is saved

**Actual behavior**

Error occurs: 
```
admission webhook "validator.harvesterhci.io" denied the request: NFS backup target should not have bucket name or region
```
<img width="1443" alt="Screenshot 2022-01-07 at 11 49 23" src="https://user-images.githubusercontent.com/3813921/148533178-c4b426b9-0832-4ec3-bb19-2c0be1969f46.png">

**Comment by w13915984028** [comment]
Reproduced in master-head ISO.

The workaround is: 
  edit setting, "Use the default value", save, the current S3 setting will be reset. 
  then, edit setting again, set it to NFS, it will be OK. 

![image](https://user-images.githubusercontent.com/31133476/148564469-cb037b23-9ad1-4b89-b961-a28e1a2b9ee4.png)

**Comment by w13915984028** [comment]
When the backup target is NFS at first, then directly set it to a valid S3 is fine:

![image](https://user-images.githubusercontent.com/31133476/148565319-fba4d1a3-78ac-463c-9b10-8c3cb85974ad.png)

**Comment by w13915984028** [comment]
Root cause:
When change backup target from S3 to NFS, those S3 related fields like bucket-name is kept and sent to backend, webhook blocks it.
```
 type    endpoint                                      bucket-name       bucket-region 
 {nfs nfs://192.168.122.85:/nfs_server/nfs_share   harvester-ci-testing us-east-2  true}
```
Will discuss where to clear those unrelated fields.

```
time="2022-01-07T16:46:57Z" level=error msg="validate failed with:&{nfs nfs://192.168.122.85:/nfs_server/nfs_share   harvester-ci-testing us-east-2  true}"

time="2022-01-07T16:46:57Z" level=debug msg="Request (user: user-c7wjs, harvesterhci.io/v1beta1, Kind=Setting, namespace: , name: backup-target, operation: UPDATE) operation is rejected: NFS backup target should not have bucket name or region"
```

**Comment by noahgildersleeve** [comment]
Tested in master-ebde990d-head. 
### Verification steps
1. Create a backup target with a valid S3 target
2. Save
3. Change target to a valid NFS address
4. Save
5. Verify there are no errors.

**Comment by dseralathan** [comment]
Geeting below error

admission webhook "validator.harvesterhci.io" denied the request: Can't update the backup target. There is a VM either in the process of creating or deleting

But there is no VM is that state .

**Comment by albertkohl-monotek** [comment]
I think I remember that being a thing fixed in v1.1.1 as well, but might be wrong. You can't clear or delete the backup target. Also I vaguely remember an issue where if you make the backup target unavailable, it breaks things elsewhere, can't remember what though.
